src/io.c

```diff
@@ -84,18 +86,23 @@ static struct gpio_callback button_0_cb_data;
 static struct gpio_callback button_1_cb_data;
 
 void button_0_pressed(struct device *dev, struct gpio_callback *cb, uint32_t pins){
+    #ifndef NATIVE_POSIX
```

Comment:
@hendrick404 Could you please also move this to the outer scope of the file, please? (as in display.c)